src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -87,7 +94,7 @@ export class OsdListComponent implements OnInit {
     this.osdService.getList().subscribe((data: any[]) => {
       this.osds = data;
       data.map((osd) => {
-        osd.collectedStates = this.collectStates(osd);
+        osd.collectedStates = OsdListComponent.collectStates(osd);
```

Comment:
Making a method static if possible has the following advantages:

- You can be sure that there are no side-effects regarding the instance of a class
- Easier to test as you don't need an instance
- Not relevant here, but it can be used outside the class without constructing an instance
- You can see that it's not used in the template, as static methods cannot directly be used in the template